tests/src/JIT/Directed/VectorABI/VectorMgdMgd.cs

```diff
@@ -49,166 +244,659 @@ public HVATests()
+        {
+            checkValues("test1Argument_HVA64_03(arg1.v0)", arg1.v0, 0);
+            checkValues("test1Argument_HVA64_03(arg1.v1)", arg1.v1, 1 * Vector64<T>.Count);
+            checkValues("test1Argument_HVA64_03(arg1.v2)", arg1.v2, 2 * Vector64<T>.Count);
+        }
+
+        public void Init_Reflection_Args_HVA64_03()
+        {
+            isReflection = true;
+            reflectionParameterTypes = new Type[] { typeof(HVA64_03) };
+            reflectionMethodInfo = typeof(HVATests<T>).GetMethod(nameof(HVATests<T>.test1Argument_HVA64_03), reflectionParameterTypes);
+            reflectionInvokeArgs = new object[] { hva64_03 };
+        }
+
+        //====================   Tests for passing 1 argumnet of HVA64_04
+
+        // Test the case where we've passed in a single argument HVA of 4 vectors.
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public void test1Argument_HVA64_04(HVA64_04 arg1)
+        {
+            checkValues("test1Argument_HVA64_04(arg1.v0)", arg1.v0, 0);
+            checkValues("test1Argument_HVA64_04(arg1.v1)", arg1.v1, 1 * Vector64<T>.Count);
+            checkValues("test1Argument_HVA64_04(arg1.v2)", arg1.v2, 2 * Vector64<T>.Count);
+            checkValues("test1Argument_HVA64_04(arg1.v3)", arg1.v3, 3 * Vector64<T>.Count);
+        }
+
+        public void Init_Reflection_Args_HVA64_04()
         {
-            checkValues(s.v0, 0);
-            checkValues(s.v1, Vector128<T>.Count);
+            isReflection = true;
+            reflectionParameterTypes = new Type[] { typeof(HVA64_04) };
+            reflectionMethodInfo = typeof(HVATests<T>).GetMethod(nameof(HVATests<T>.test1Argument_HVA64_04), reflectionParameterTypes);
+            reflectionInvokeArgs = new object[] { hva64_04 };
         }
 
+        //====================   Tests for passing 1 argumnet of HVA64_05
+
+        // Test the case where we've passed in a single argument HVA of 5 vectors.
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public void test1Argument_HVA64_05(HVA64_05 arg1)
+        {
+            checkValues("test1Argument_HVA64_05(arg1.v0)", arg1.v0, 0);
+            checkValues("test1Argument_HVA64_05(arg1.v1)", arg1.v1, 1 * Vector64<T>.Count);
+            checkValues("test1Argument_HVA64_05(arg1.v2)", arg1.v2, 2 * Vector64<T>.Count);
+            checkValues("test1Argument_HVA64_05(arg1.v3)", arg1.v3, 3 * Vector64<T>.Count);
+            checkValues("test1Argument_HVA64_05(arg1.v4)", arg1.v4, 4 * Vector64<T>.Count);
+        }
+
+        public void Init_Reflection_Args_HVA64_05()
+        {
+            isReflection = true;
+            reflectionParameterTypes = new Type[] { typeof(HVA64_05) };
+            reflectionMethodInfo = typeof(HVATests<T>).GetMethod(nameof(HVATests<T>.test1Argument_HVA64_05), reflectionParameterTypes);
+            reflectionInvokeArgs = new object[] { hva64_05 };
+        }
+
+        //===============   Tests for return values if HFA64
```

Comment:
nit: "of HVA64"

Reply:
I believe that up to 8 registers are used to pass HVA or HFA:

2.	If the argument is an HFA or an HVA and there are sufficient unallocated SIMD and Floating-point registers (NSRN + number of members ≤ 8), then the argument is allocated to SIMD and Floating-point Registers (with one register per member of the HFA or HVA).  The NSRN is incremented by the number of registers used.  The argument has now been allocated.

Reply:
Right, but for the HFAs and short vector HVAs, it explicitly says that to be considered a homogeneous aggregate it must have no more than 4 members (each) so that although more than one might be passed in registers, each one can have no more than 4 members, but it doesn't say that about the SVAs (as far as I could find).